cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -613,13 +626,19 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_BENCHMARKS)
     set(GTEST_CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
```

Pass linkers to the gtest cmake. This might fix the trusty failure maybe:

```suggestion
     set(GTEST_CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                                                  -DCMAKE_AR=${CMAKE_AR}
                                                  -DCMAKE_RANLIB=${CMAKE_RANLIB}
```